tests/integration/ceph_mgr_test.go

```diff
@@ -0,0 +1,216 @@
+}
+
+func (suite *MgrSmokeSuite) TestStatus() {
+	logger.Infof("Testing .... <orch status>")
+	err, status := suite.CLI.Execute([]string{"status"})
+	assert.Nil(suite.T(), err)
+	assert.Equal(suite.T(), status, "Backend: rook\nAvailable: True")
+}
+
+func (suite *MgrSmokeSuite) TestHostLs() {
+	logger.Infof("Testing .... <orch host ls>")
+	// Get the orchestrator hosts
+	err, output := suite.CLI.Execute([]string{"host", "ls", "json"})
+	assert.Nil(suite.T(), err)
+
+	hosts := []byte(output)
+	var hostsList []host
+
+	err = json.Unmarshal(hosts, &hostsList)
+	if err != nil {
+		assert.Nil(suite.T(), err)
+	}
+
+	var hostOutput []string
+	for _, hostItem := range hostsList {
+		hostOutput = append(hostOutput, hostItem.Addr)
+	}
+	sort.Strings(hostOutput)
+
+	// get the k8s nodes
+	nodes, err := suite.k8sh.Kubectl("get", "nodes", "-o=custom-columns=NAME:.metadata.name", "--no-headers")
+	assert.Nil(suite.T(), err)
+	k8sNodes := strings.Fields(nodes)
+	sort.Strings(k8sNodes)
+
+	// nodes anf hosts must be the same
+	assert.Equal(suite.T(), hostOutput, k8sNodes)
+
+}
+
+func (suite *MgrSmokeSuite) TestCreateOSD() {
+	logger.Infof("Testing .... <orch create OSD>")
+
+	// Get the first available device
+	err, deviceList := suite.CLI.Execute([]string{"device", "ls"})
+	assert.Nil(suite.T(), err)
+
+	inventory := make([]map[string]interface{}, 0)
+
+	err = json.Unmarshal([]byte(deviceList), &inventory)
+	assert.Nil(suite.T(), err)
+
+	selectedNode := ""
+	selectedDevice := ""
+	for _, node := range inventory {
+		for _, device := range node["devices"].([]interface{}) {
+			if device.(map[string]interface{})["available"].(bool) {
+				selectedNode = node["name"].(string)
+				selectedDevice = device.(map[string]interface{})["id"].(string)
+				break
+			}
+		}
+		if selectedDevice != "" {
+			break
+		}
+	}
+	assert.NotEqual(suite.T(), "", selectedDevice, "No devices available to create test OSD")
+	assert.NotEqual(suite.T(), "", selectedNode, "No nodes available to create test OSD")
+
+	if selectedDevice == "" || selectedNode == "" {
+		return
+	}
+	// Create the OSD
+	err, _ = suite.CLI.Execute([]string{"osd", "create", fmt.Sprintf("%s:%s", selectedNode, selectedDevice)})
+	assert.Nil(suite.T(), err)
+
+	err = suite.k8sh.WaitForPodCount("app=rook-ceph-osd", suite.namespace, 1)
+	assert.Nil(suite.T(), err)
+
+}
+
+func (suite *MgrSmokeSuite) TestServiceLs() {
+	logger.Info("Testing .... <orch ls>")
+	err, output := suite.CLI.Execute([]string{"ls", "json"})
+	assert.Nil(suite.T(), err)
+
+	services := []byte(output)
+	var servicesList []service
+
+	err = json.Unmarshal(services, &servicesList)
+	assert.Nil(suite.T(), err)
+
+	nameSpace := fmt.Sprintf("--namespace=%s", suite.namespace)
+	for _, svc := range servicesList {
+		labelFilter := fmt.Sprintf("-l app=rook-ceph-%s", svc.Service_name)
+		pods, err := suite.k8sh.Kubectl("get", "pods", labelFilter, nameSpace, "-o=custom-columns=NAME:.metadata.name", "--no-headers")
```

Comment:
The helper `k8sutil.PodsRunningWithLabel()` may be helpful here instead of the kubectl command